File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/FragmentConnection.java
package com.h2Invent.skibin;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.json.JSONObject;

public class FragmentConnection extends Fragment {
    public static final String TITLE = "Verbindung";

    private JSONObject jsonRes;
    private static TextView userName;
    private static TextView userEmail;
    private static TextView userOrg;
    private static Button saveUser;
    private static Button resetUser;
    private static EditText userEmailCode;
    private static Button userEmailBtn;
    private static LinearLayout userinfo;
    public static LinearLayout userEmailLayout;
    public static Button scanConnection;
    private static String sUserName;
    private static String sUserOrg;
    private static String sUSerEmail;
    private static boolean isUser;



    private static String sOrgPartner;
    private static String sOrgName;
    private static boolean isOrg;
    private static LinearLayout orgConnection;
    private static LinearLayout userConnection;
    private static Button orgSave;
    private static Button orgReset;
    private static LinearLayout orgInfo;
    private static TextView orgName;
    private static TextView orgPartner;


    public static FragmentConnection newInstance() {

        return new FragmentConnection();
    }


    private OnItemSelectedListener listener;

    public interface OnItemSelectedListener {
        public void scanClicked();

        public void userSaveClicked();
        public void orgSaveClicked();

        public void resetClicked();

        public void userEmailConfirmClicked(String Token);
        public void init();
    }
    // Store the listener (activity) that will have events fired once the fragment is attached

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        if (context instanceof OnItemSelectedListener) {
            listener = (OnItemSelectedListener) context;
        } else {
            throw new ClassCastException(context.toString()
                    + " must implement MyListFragment.OnItemSelectedListener");
        }

    }

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_connection, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        scanConnection = (Button) view.findViewById(R.id.scanConnection);

        userinfo = (LinearLayout) view.findViewById(R.id.user_info);
        userName = (TextView) view.findViewById(R.id.user_name);
        userEmail = (TextView) view.findViewById(R.id.user_email);
        userOrg = (TextView) view.findViewById(R.id.user_organisation);
        userEmailLayout = (LinearLayout) view.findViewById(R.id.userEmailCOnfirmation);
        userEmailCode = (EditText) view.findViewById(R.id.edituseremailCode);
        userEmailBtn = (Button) view.findViewById(R.id.buttonEmailCOnfirmation);
        resetUser = (Button) view.findViewById(R.id.resetUserConnection);
        saveUser = (Button) view.findViewById(R.id.saveUserConnection);

        orgReset = (Button) view.findViewById(R.id.connectionResetOrg);
        orgSave = (Button) view.findViewById(R.id.connectionOrgSave);
        orgName = (TextView) view.findViewById(R.id.org_name);
        orgPartner = (TextView) view.findViewById(R.id.orgPartner);
        orgInfo = (LinearLayout) view.findViewById(R.id.orgInfo);

        if(isUser){
            orgInfo.setVisibility(View.GONE);
            userinfo.setVisibility(View.VISIBLE);
            userName.setText(sUserName);
            userOrg.setText(sUserOrg);
            userEmail.setText(sUSerEmail);
        }
        if (isOrg){
            userinfo.setVisibility(View.GONE);
            orgInfo.setVisibility(View.VISIBLE);
            orgName.setText(sOrgName);
            orgPartner.setText(sOrgPartner);
        }

        saveUser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.userSaveClicked();
            }
        });
        orgSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.orgSaveClicked();
            }
        });
        scanConnection.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.scanClicked();
            }
        });
        resetUser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.resetClicked();
            }
        });
        orgReset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.resetClicked();
            }
        });
        userEmailBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.userEmailConfirmClicked(userEmailCode.getText().toString());
            }
        });
        listener.init();
    }

    public void setUserText(String name, String organisation, String email) {
        userName.setText(name);
        userOrg.setText(organisation);
        userEmail.setText(email);
        userinfo.setVisibility(View.VISIBLE);
    }

    public void activateEmail(boolean in) {
        userEmailLayout.setVisibility(in?View.VISIBLE:View.GONE);
        scanConnection.setEnabled(!in);
        scanConnection.setText("Prüfen Sie Ihre E-Mails");

    }

    public void activateScan(){
        scanConnection.setEnabled(true);
    }
    public void deactivateScan(){
        scanConnection.setEnabled(false);
    }
    public void activateSave(){
        saveUser.setEnabled(true);
    }
    public void deactivateSave(){
        saveUser.setEnabled(false);
    }
    public void activateReset(){
        resetUser.setEnabled(true);
    }
    public void deactivateReset(){
        resetUser.setEnabled(false);
    }
    public void setScanText(String text){
        scanConnection.setText(text);
    }

    public String getsUserName() {
        return sUserName;
    }

    public void setsUserName(String sUserName) {
        this.sUserName = sUserName;
    }

    public String getsOrg() {
        return sUserOrg;
    }

    public void setsOrg(String sOrg) {
        this.sUserOrg = sOrg;
    }

    public String getsEmail() {
        return sUSerEmail;
    }

    public void setsEmail(String sEmail) {
        this.sUSerEmail = sEmail;
    }

    public boolean isUser() {
        return isUser;
    }

    public void setUser(boolean user) {
        isUser = user;
    }
    public static String getsOrgPartner() {
        return sOrgPartner;
    }

    public static void setsOrgPartner(String sOrgPartner) {
        FragmentConnection.sOrgPartner = sOrgPartner;
    }

    public static String getsOrgName() {
        return sOrgName;
    }

    public static void setsOrgName(String sOrgName) {
        FragmentConnection.sOrgName = sOrgName;
    }

    public static boolean isIsOrg() {
        return isOrg;
    }

    public static void setIsOrg(boolean isOrg) {
        FragmentConnection.isOrg = isOrg;
    }
    public void setOrgText(String name, String partner){
        orgName.setText(name);
        orgPartner.setText(partner);
        orgInfo.setVisibility(View.VISIBLE);
    }
    public void enableOrgSetting(boolean active){
        orgSave.setEnabled(active);
        orgSave.setVisibility(active?View.VISIBLE:View.GONE);
        orgReset.setEnabled(active);
        orgReset.setVisibility(active?View.VISIBLE:View.GONE);
    }
    public void enableUSerSetting(boolean active){
        saveUser.setEnabled(active);
        saveUser.setVisibility(active?View.VISIBLE:View.GONE);
        resetUser.setEnabled(active);
        resetUser.setVisibility(active?View.VISIBLE:View.GONE);
    }
    public void showUserInfo(boolean in){
        userinfo.setVisibility(in?View.VISIBLE:View.GONE);
    }
    public void showOrgInfo(boolean in){
        orgInfo.setVisibility(in?View.VISIBLE:View.GONE);
    }
    public void enableScanBtn(boolean in){
        scanConnection.setEnabled(in);
    }
    public void enableUSerComplete(boolean in){
        enableUSerSetting(in);
        activateEmail(in);
    }
}

File: app/src/main/java/com/h2Invent/skibin/FragmentConnection.kt
package com.h2Invent.skibin

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.google.android.material.button.MaterialButton

class FragmentConnection : Fragment() {
    private var listener: OnItemSelectedListener? = null

    private lateinit var userInfo: LinearLayout
    private lateinit var userName: TextView
    private lateinit var userEmail: TextView
    private lateinit var userOrg: TextView
    private lateinit var saveUser: MaterialButton
    private lateinit var resetUser: MaterialButton
    private lateinit var userEmailLayout: LinearLayout
    private lateinit var userEmailCode: EditText
    private lateinit var userEmailButton: MaterialButton
    private lateinit var scanConnection: MaterialButton

    private lateinit var orgSave: MaterialButton
    private lateinit var orgReset: MaterialButton
    private lateinit var orgInfo: LinearLayout
    private lateinit var orgName: TextView
    private lateinit var orgPartner: TextView

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = context as? OnItemSelectedListener
            ?: error("Parent activity must implement OnItemSelectedListener")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_connection, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        scanConnection = view.findViewById(R.id.scanConnection)
        userInfo = view.findViewById(R.id.user_info)
        userName = view.findViewById(R.id.user_name)
        userEmail = view.findViewById(R.id.user_email)
        userOrg = view.findViewById(R.id.user_organisation)
        saveUser = view.findViewById(R.id.saveUserConnection)
        resetUser = view.findViewById(R.id.resetUserConnection)
        userEmailLayout = view.findViewById(R.id.userEmailCOnfirmation)
        userEmailCode = view.findViewById(R.id.edituseremailCode)
        userEmailButton = view.findViewById(R.id.buttonEmailCOnfirmation)
        orgSave = view.findViewById(R.id.connectionOrgSave)
        orgReset = view.findViewById(R.id.connectionResetOrg)
        orgInfo = view.findViewById(R.id.orgInfo)
        orgName = view.findViewById(R.id.org_name)
        orgPartner = view.findViewById(R.id.orgPartner)

        scanConnection.setOnClickListener { listener?.scanClicked() }
        saveUser.setOnClickListener { listener?.userSaveClicked() }
        resetUser.setOnClickListener { listener?.resetClicked() }
        orgSave.setOnClickListener { listener?.orgSaveClicked() }
        orgReset.setOnClickListener { listener?.resetClicked() }
        userEmailButton.setOnClickListener { listener?.userEmailConfirmClicked(userEmailCode.text?.toString().orEmpty()) }

        listener?.init()
    }

    fun setUserText(name: String, organisation: String, email: String) {
        userName.text = name
        userOrg.text = organisation
        userEmail.text = email
        showUserInfo(true)
    }

    fun setOrgText(name: String, partner: String) {
        orgName.text = name
        orgPartner.text = partner
        showOrgInfo(true)
    }

    fun activateEmail(visible: Boolean) {
        userEmailLayout.visibility = if (visible) View.VISIBLE else View.GONE
        scanConnection.isEnabled = !visible
        if (visible) scanConnection.text = getString(R.string.emailCheckHint)
    }

    fun setScanText(text: String) {
        scanConnection.text = text
    }

    fun enableOrgSetting(active: Boolean) {
        orgSave.isEnabled = active
        orgSave.visibility = if (active) View.VISIBLE else View.GONE
        orgReset.isEnabled = active
        orgReset.visibility = if (active) View.VISIBLE else View.GONE
    }

    fun enableUserSetting(active: Boolean) {
        saveUser.isEnabled = active
        saveUser.visibility = if (active) View.VISIBLE else View.GONE
        resetUser.isEnabled = active
        resetUser.visibility = if (active) View.VISIBLE else View.GONE
    }

    fun showUserInfo(visible: Boolean) {
        userInfo.visibility = if (visible) View.VISIBLE else View.GONE
    }

    fun showOrgInfo(visible: Boolean) {
        orgInfo.visibility = if (visible) View.VISIBLE else View.GONE
    }

    fun enableScanButton(enabled: Boolean) {
        scanConnection.isEnabled = enabled
    }

    interface OnItemSelectedListener {
        fun scanClicked()
        fun userSaveClicked()
        fun orgSaveClicked()
        fun resetClicked()
        fun userEmailConfirmClicked(token: String)
        fun init()
    }

    companion object {
        const val TITLE = "Verbindung"
        fun newInstance() = FragmentConnection()
    }
}
